### PR TITLE
Bug fixes *Needs Core Recompile*

### PR DESCRIPTION
**Pull request by kevin-10**
- Combatant no works via props Gump
- Fixed valid items showing on invalid internal map
- Healing with bandages (not vet) can now be used on Despise Creatures
- Animal Lore now shows the proper regen rates, as well as EC character sheet
- Pets now gain skill above the total skill cap
- Summons now use the proper control slots
- Deepwater fishing removed from TerMur
- Creatures that naturally poisons now start with poisoning magical ability
- Added Armor Pierce weapon ability to various creatures that was erroneously omitted
- Bard songs no longer flag the caster for criminal beneficial actions
- Bard songs no longer give beneficial actions that would otherwise flag the caster criminal
- Fixed Combat Training
- Spellweaving creatures now have a natural arcane circle, giving it +1 strength per 20 spellweaving Skill
- AddonComponents with null addon should now delete when a house is demolished
- Separated Damage Packet from Mobile.cs to a separate method. This was needed and provides more flexibility as to how damage can be shown.

### CODE REVIEW

File: Scripts/Mobiles/Normal/ClockworkScorpion.cs [hunk 1]
```diff
@@ -11,35 +11,35 @@ public class ClockworkScorpion : BaseCreature, IRepairableMobile
...
-            this.SetSkill(SkillName.Wrestling, 50.1, 80.0);
+            SetSkill(SkillName.MagicResist, 30.1, 50.0);
+            SetSkill(SkillName.Poisoning, 95.1, 100.0);
+            SetSkill(SkillName.Tactics, 70.1, 90.0);
+            SetSkill(SkillName.Wrestling, 50.1, 80.0);
 
-            this.Fame = 3500;
-            this.Karma = -3500;
+            Fame = 3500;
+            Karma = -3500;
```
Comment (Argalep) [inline Scripts/Mobiles/Normal/ClockworkScorpion.cs]: missing SetMagicalAbility(MagicalAbility.Poisoning);

File: Scripts/Mobiles/Normal/GiantSpider.cs [hunk 2]
```diff
@@ -10,39 +10,39 @@ public class GiantSpider : BaseCreature
...
 
-            this.VirtualArmor = 16;
+            VirtualArmor = 16;
 
-            this.Tamable = true;
-            this.ControlSlots = 1;
-            this.MinTameSkill = 59.1;
+            Tamable = true;
+            ControlSlots = 1;
+            MinTameSkill = 59.1;
```
Comment (Argalep) [inline Scripts/Mobiles/Normal/GiantSpider.cs]: missing SetMagicalAbility(MagicalAbility.Poisoning);

**Comment by Argalep** [comment]
CorrosiveSlime, DreadSpider, GiantSpider, RuneBeetle, Scorpion, Slime, Snake, WolfSpider (missing creature features) - SetMagicalAbility(MagicalAbility.Poisoning);

File: Scripts/Services/Expansions/Time Of Legends/Mobiles/Dimetrosaur.cs [hunk 3]
```diff
@@ -115,7 +115,7 @@ public Dimetrosaur(Serial serial)
         public override void Serialize(GenericWriter writer)
         {
             base.Serialize(writer);
-            writer.Write(1);
+            writer.Write(3);
```
Comment (Argalep) [inline Scripts/Services/Expansions/Time Of Legends/Mobiles/Dimetrosaur.cs]: 2 should be

File: Scripts/Mobiles/Normal/GiantSpider.cs (hunk 2 above)
Comment (kevin-10) [reply Scripts/Mobiles/Normal/GiantSpider.cs]: No, its set in basecreature. Adding it during versioning was for pre-existing creatures.

File: Scripts/Services/Expansions/Time Of Legends/Mobiles/Dimetrosaur.cs (hunk 3 above)
Comment (kevin-10) [reply Scripts/Services/Expansions/Time Of Legends/Mobiles/Dimetrosaur.cs]: Wouldn't impact anything, but I'll change it to 2.

**Pull request merged**